tests/libvmi/factory.go

```diff
@@ -40,20 +40,29 @@ func NewFedora(opts ...Option) *kvirtv1.VirtualMachineInstance {
 	return New(opts...)
 }
 
-// NewCirros instantiates a new CirrOS based VMI configuration
-func NewCirros(opts ...Option) *kvirtv1.VirtualMachineInstance {
+func newCirros(containerDisk string, opts ...Option) *kvirtv1.VirtualMachineInstance {
 	// Supplied with no user data, Cirros image takes 230s to allow login
 	withNonEmptyUserData := WithCloudInitNoCloudUserData("#!/bin/bash\necho hello\n", true)
 
 	cirrosOpts := []Option{
-		WithContainerImage(cd.ContainerDiskFor(cd.ContainerDiskCirros)),
+		WithContainerImage(containerDisk),
 		withNonEmptyUserData,
 		WithResourceMemory(cirrosMemory()),
 	}
 	cirrosOpts = append(cirrosOpts, opts...)
 	return New(cirrosOpts...)
 }
 
+// NewCirros instantiates a new CirrOS based VMI configuration
+func NewCirros(opts ...Option) *kvirtv1.VirtualMachineInstance {
+	return newCirros(cd.ContainerDiskFor(cd.ContainerDiskCirros), opts...)
+}
+
+// NewCirrosCustomLocation instantiates a new CirrOS custom location based VMI configuration
+func NewCirrosCustomLocation(opts ...Option) *kvirtv1.VirtualMachineInstance {
```

Comment:
I saw only one single place it is used. I do not think it is worth adding under `libvmi`.
Specialized options that are rarely used and have some logic, should better be placed close to its usage.

Reply:
Removed it, thank you.